### PR TITLE
336390 waste input for last year

### PR DESCRIPTION
**Pull request by AtosSaqlain**

### CODE REVIEW

File: src/EPR.Accreditation.Facade.Common/Dtos/Portal/MaterialWasteOutputsDto.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace EPR.Accreditation.Facade.Common.Dtos.Portal
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade.Common/Dtos/Portal/MaterialWasteOutputsDto.cs:1]: What's this one all about? We already have a MaterialOutputDto that covers all these properties

File: src/EPR.Accreditation.Facade.Common/RESTservices/HttpAccreditationService.cs [hunk 2]
```diff
@@ -90,6 +90,13 @@ public async Task<Guid> CreateAccreditation(Dtos.Accreditation accreditation)
             return await Get<List<AccreditationTaskProgress>>($"{accreditationExternalId}/TaskProgress");
         }
 
+        public async Task<AccreditationMaterial> GetLastCalendarYearWaste(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade.Common/RESTservices/HttpAccreditationService.cs:93]: There's already functionality that does waste last year: AccreditationMaterialService.GetReprocessedWasteLastYear

File: src/EPR.Accreditation.Facade.Common/RESTservices/Interfaces/IHttpAccreditationService.cs [hunk 3]
```diff
@@ -33,5 +33,6 @@ public interface IHttpAccreditationService
 
         Task<List<Dtos.AccreditationTaskProgress>> GetTaskProgress(
             Guid accreditationExternalId);
+        Task<AccreditationMaterial> GetLastCalendarYearWaste(Guid accreditationExternalId, Guid accreditationMaterialExternalId);
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade.Common/RESTservices/Interfaces/IHttpAccreditationService.cs:36]: As above

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs [hunk 4]
```diff
@@ -105,5 +106,17 @@ public async Task<IActionResult> GetHasPermitExemption(Guid accreditationExterna
 
             return Ok(taskProgress);
         }
+
+        [HttpGet("LastCalendarYearWaste")]
+        public async Task<IActionResult> LastCalendarYearWaste(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs:114]: As above

File: src/EPR.Accreditation.Facade/Profiles/AccreditationProfile.cs [hunk 5]
```diff
@@ -13,10 +13,27 @@ public AccreditationProfile()
                 .ForMember(d => d.Contaminents, o => o.MapFrom(s => s.TonnesContaminents))
                 .ForMember(d => d.ProcessLoss, o => o.MapFrom(s => s.TonnesProcessLoss));
 
+            CreateMap<MaterialWasteOutputsDto, MaterialReprocessorDetails>()
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Profiles/AccreditationProfile.cs:16]: I suspect most of this is already covered too

File: src/EPR.Accreditation.Facade.Common/Dtos/Portal/MaterialWasteOutputsDto.cs (hunk 1 above)
Comment (AtosSaqlain) [reply src/EPR.Accreditation.Facade.Common/Dtos/Portal/MaterialWasteOutputsDto.cs:1]: The new Dto contains a different set of properties, although, both contain [WasteLastYear]

File: src/EPR.Accreditation.Facade/Profiles/AccreditationProfile.cs (hunk 5 above)
Comment (AtosSaqlain) [reply src/EPR.Accreditation.Facade/Profiles/AccreditationProfile.cs:16]: Please see my response to the first comment above.

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**